packages/react/src/primitives/utils/getFieldErrorMessageId.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Returns a string combining `errorMessageId` with `-error` if `hasError` is true and `errorMessageId` is provided; otherwise, returns undefined.
+ *
+ * @param {string} errorMessageId - The base ID to be used for creating the error message ID.
+ * @param {boolean} [hasError=false] - Indicates whether there is an error. Defaults to false.
+ * @returns {string | undefined} The combined ID string if `hasError` is true and `errorMessageId` is provided, otherwise `undefined`.
+ */
+export const getFieldErrorMessageId = (
```

Comment:
Are `getFieldDescriptionId` and `getFieldErrorMessageId` different enough to each have a distinct util? Could they be combined? With an empty string having a falsy value, could we instead check to see if the provided value (`hasError` or `descriptiveText`) evaluates to false, then conditionally append a provided string to `stableId`?

Reply:
On second thought, since these functions operate very similarly to the existing `useTestId()`, would it make sense to instead make that exiting util more generalized in its description and use it for these two cases?

Something like:
```
export const useUniqueComponentId = (
  id?: string,
  component?: string
): string | undefined =>
  id && component ? `${id}-${component}` : undefined;
```
which could be used as `getUniqueComponentId(stableId, descriptiveText ? 'description' : undefined)` and `getUniqueComponentId(stableId, hasError ? 'error' : undefined)`?